storage/storage-interface/src/finality_view.rs

```diff
@@ -0,0 +1,159 @@
+            None,
+        );
+        // Sanity checks: finalization should not be set on an empty database,
+        // the finality version should not exceed the latest committed.
+        match self.reader.get_latest_state_checkpoint_version()? {
+            None => return Err(AptosDbError::Other("no ledger states to finalize".into())),
+            Some(ver) => {
+                if end_ver > ver {
+                    return Err(AptosDbError::Other(format!(
+                        "finality version {end_ver} exceeds committed version {ver}"
+                    )));
+                }
+            },
+        }
+        let ledger_info = LedgerInfo::new(block_info, HashValue::zero()); // we don't use consensus_data_hash
+        let aggregate_signature = aggregate_signature::AggregateSignature::empty(); // we don't use
+                                                                                    // aggregate_signatures
+        let ledger_info = LedgerInfoWithSignatures::new(ledger_info, aggregate_signature);
+        let mut fin_legder_info = self.finalized_ledger_info.write().unwrap();
+        *fin_legder_info = Some(ledger_info);
+        Ok(())
+    }
+}
+
+impl<Db: DbReader> DbReader for FinalityView<Db> {
+    fn get_read_delegatee(&self) -> &dyn DbReader {
+        &self.reader
+    }
+
+    fn get_latest_ledger_info_option(&self) -> Result<Option<LedgerInfoWithSignatures>> {
+        let fin_ledger_info = self.finalized_ledger_info.read().unwrap();
+        Ok(fin_ledger_info.clone())
+    }
+
+    fn get_latest_version(&self) -> Result<Version> {
+        let fin_ledger_info = self.finalized_ledger_info.read().unwrap();
+        fin_ledger_info
+            .as_ref()
+            .map(|li| li.ledger_info().version())
+            .ok_or_else(|| AptosDbError::NotFound("finalized version".into()))
+    }
+
+    fn get_latest_state_checkpoint_version(&self) -> Result<Option<Version>> {
+        let fin_ledger_info = self.finalized_ledger_info.read().unwrap();
+        let version = fin_ledger_info
+            .as_ref()
+            .map(|li| li.ledger_info().version());
+        Ok(version)
+    }
+
+    // TODO: override any other methods needed to maintain the illusion.
+}
+
+#[cfg(test)]
+mod tests {
+    use aptos_types::{aggregate_signature::AggregateSignature, ledger_info::LedgerInfo};
+
+    use super::*;
+    use crate::mock::MockDbReaderWriter;
+
+    #[test]
+    fn test_get_latest_ledger_info() {
+        // If the mock is changed to be stateful, this should be ref-counted
+        // and shared with the view.
+        let mock = MockDbReaderWriter;
+        let view = FinalityView::new(MockDbReaderWriter);
+
+        let ledger_info = view.get_latest_ledger_info_option().unwrap();
+        assert_eq!(ledger_info, None);
+        let blockheight = 1;
+
+        // Set the finalized ledger info
+        view.set_finalized_block_height(blockheight).unwrap();
+
+        // Capture the block event once
```

Comment:
Do we know how the `BlockEvent` is used by the API precisely--if at all?

Reply:
There is [a comment](https://github.com/movementlabsxyz/aptos-core/blob/ea91067b81f9673547417bff9c70d5a2fe1b0e7b/storage/aptosdb/src/db/include/aptosdb_reader.rs#L663) stating the developers would rather return `BlockInfo` from this method, but we make do with information from this structure.